feat: re-enable surplus modal pop up

alfetopito opened the pull request:
# Summary

- Re-enable surplus modal pop-up when progress bar is closed
- Remove progress bar from activity modal
- Add button to bring progress bar back/focus on running progress bar

# To Test

1. Place order, don't close modal
2. Open activity modal
* There should be an ugly `show progress` button
3. Click on `show progress`
* Should close the activity modal and existing progress bar should be running
4. Close progress bar
5. Open activity modal
6. Click on `show progress`
* Should close activity modal and show progress bar in another pop-up modal
* Everything in this modal should work like regular modal (cancel button, expired link, etc)
7. Place order with high slippage (50%) for making sure there'll be surplus
8. Close progress bar and wait for it to trade
* When traded, progress bar should pop-up in a new modal in the last step

Additionally, do a full regression test on progress bar up to here, as with the refactoring some things might have been broken.

alfetopito commented on apps/cowswap-frontend/src/modules/account/containers/Transaction/ActivityDetails.tsx:
```diff
@@ -203,7 +205,19 @@ export function ActivityDetails(props: {
   const hideCustomRecipientWarning = useHideReceiverWalletBanner()
   const orderProgressBarV2Props = useOrderProgressBarV2Props({ activityDerivedState, chainId })
-  const surplusData = useGetSurplusData(order)
+  const setShowProgressBar = useAddOrderToSurplusQueue() // TODO: not exactly the proper tool, rethink this
```
I'm re-using the logic for triggering the surplus modal.
Not happy with this, but it's working.

In the good to haves to refactor it.

alfetopito commented on apps/cowswap-frontend/src/modules/account/containers/Transaction/ActivityDetails.tsx:
```diff
@@ -393,19 +407,11 @@ export function ActivityDetails(props: {
       </Summary>
 
       <EthFlowStepper order={order} extend={!!orderProgressBarV2Props} />
-      {showProgressBar && orderProgressBarV2Props && orderProgressBarV2Props.stepName !== 'finished' && (
-        <OrderProgressBarV2
-          {...orderProgressBarV2Props}
-          order={order}
-          surplusData={surplusData}
-          chainId={chainId}
-          receiverEnsName={receiverEnsName || undefined}
-        />
-      )}
```
Removed progress bar completely from activity modal

alfetopito commented on apps/cowswap-frontend/src/modules/account/containers/Transaction/StatusDetails.tsx:
```diff
@@ -130,6 +131,8 @@ export function StatusDetails(props: StatusDetailsProps) {
         </StatusLabelBelow>
       )}
+      {/* TODO: Probably not the right component, just placeholder for now */}
+      {showProgressBar && <CancelButton onClick={showProgressBar}>Show progress</CancelButton>}
```
Very ugly, I know. I hope @fairlighteth will have a better idea where to put and how to display this.
![image](https://github.com/user-attachments/assets/0ee41b47-5a28-40b4-985a-0dded0fc11f5)

fairlighteth replied:
Will review

alfetopito commented on apps/cowswap-frontend/src/modules/swap/containers/ConfirmSwapModalSetup/index.tsx:
```diff
@@ -155,46 +152,55 @@ export function ConfirmSwapModalSetup(props: ConfirmSwapModalSetupProps) {
+}
+
+// TODO: doesn't belong here
+export function useSetupAdditionalProgressBarProps(chainId: SupportedChainId, order: Order | undefined) {
```
TODO: need to move these 2 hooks out

alfetopito commented on apps/cowswap-frontend/src/modules/swap/containers/SurplusModalSetup/index.tsx:
```diff
@@ -4,33 +4,42 @@ import { useWalletInfo } from '@cowprotocol/wallet'
 import { useOrderIdForSurplusModal, useRemoveOrderFromSurplusQueue } from '../../state/surplusModal'
+import { useNavigateToNewOrderCallback, useSetupAdditionalProgressBarProps } from '../ConfirmSwapModalSetup'
 
+// TODO: rename?
 export function SurplusModalSetup() {
```
Surplus modal is not about just surplus anymore.

In the good to haves to rename/refactor this.

elena-zh commented:
Hey @alfetopito , great!

Some issues: 
1. Newly opened progress modal (called from the Account button) does not have 'View on Explorer' and 'Cancel order' buttons at the top
![canced and view](https://github.com/user-attachments/assets/ef26a86f-65e4-4c8c-aeb2-89973cb3e6d7)

2. Surplus modal (when the bar is closed, and surplus is >1USD) also does not have 'View on Explorer' link 
![image](https://github.com/user-attachments/assets/c4e687b3-72eb-4638-b784-c3c701297f36)

3. Traded instead of cancel case: 
- Place an order (set 50% slippage)
- Close the bar
- in 15-20 seconds after an order placement cancel it --> surplus modal appears. When I close the surplus modal I can see blinking 'progress bar' modal under the Surplus one (it is auto-closed, but sometimes is noticeable). 

4. case when a user is on Limit/TWAP/other content page:
- place a swap order. close the bar
- go to Limit orders page
- Open Account modal
- Press on the 'Show progress bar' button --> looks like nothing happens, the Account modal is closed, but there is no progress bar is displayed on the Limit orders page. The bar pops-up only when go back to Swap page. Confusing. 

5. Also I'm noticing a slight screen blink when an order is traded with surplus less 1USD: 
- place a tiny order
- close the bar
- wait till an order is executed and check the screen: once traded, there will be a tiny screen blink 

6. Not sure if it is an issue, when when I reopen the bar for a ETH-flow order, I don't see the stepper above. Same when a surplus modal appears for such orders
![image](https://github.com/user-attachments/assets/25f525b4-7030-4716-92a4-e2b9bb0c3c11)

7. Animation stopped working (might be an intended change?): 
- place 2 the same orders with a high price impact
- wait till the 2st order is executed --> animation is stopped on the 'Price OOM' step for the 2nd order
![image](https://github.com/user-attachments/assets/9a6f38f7-0870-4c26-8cfd-b3d61a26e328)

8. Case with 2 orders placed one after another: when the 1st order is executed with a surplus, the modal does not appear above the opened progress bar for the 2nd order
![image](https://github.com/user-attachments/assets/3926c27e-289d-43dd-9780-eb8c053cac6c)

9. Case with 2 orders placed one after another: 
- place the 1st order, close the bar
- place the 2nd order, leave the bar opened
- open Account modal
- Press on the 'Show progress' for the 1st order --> -the Account modal is closed, but I still see the progress bar for the 2nd order (not for the 1st one). WDYT about showing the 'Show progress' link to orders with closed progress bar only? 

10. This is for @fairlighteth: when Surplus modal pops up in a mobile view, some bottom elements are hidden under the Account bar 
![image](https://github.com/user-attachments/assets/85198c17-a397-4d97-9330-3af293b5301d)
![image](https://github.com/user-attachments/assets/86eed8f6-91bb-433c-9578-9ad6bd3d4bbe)

Thanks

alfetopito commented:
> Hey @alfetopito , great!
> 
> Some issues:
> 
>     1. Newly opened progress modal (called from the Account button) does not have 'View on Explorer' and 'Cancel order' buttons at the top
>        ![canced and view](https://private-user-images.githubusercontent.com/70885163/359450688-ef26a86f-65e4-4c8c-aeb2-89973cb3e6d7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQxNjU4ODcsIm5iZiI6MTcyNDE2NTU4NywicGF0aCI6Ii83MDg4NTE2My8zNTk0NTA2ODgtZWYyNmE4NmYtNjVlNC00YzhjLWFlYjItODk5NzNjYjNlNmQ3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIwVDE0NTMwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWM0MmExZjBmOWU2MWVkMTViZGJiNjU1M2JmOTEzYjBlNTZhMDRlZmFmNWM4YmFiMmYwNjMzNzQwZGM5MGQ3ZDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-1Iv5jkkWWf3EXfg5uwWcdSPBnXIQPqtzz0nrtvHfRE)

Fixed

>     2. Surplus modal (when the bar is closed, and surplus is >1USD) also does not have 'View on Explorer' link
>        ![image](https://private-user-images.githubusercontent.com/70885163/359450657-c4e687b3-72eb-4638-b784-c3c701297f36.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQxNjU4ODcsIm5iZiI6MTcyNDE2NTU4NywicGF0aCI6Ii83MDg4NTE2My8zNTk0NTA2NTctYzRlNjg3YjMtNzJlYi00NjM4LWI3ODQtYzNjNzAxMjk3ZjM2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIwVDE0NTMwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTAwMzNkODU0MmQ4YWJlY2IyZjA2YzQ5MGNkNmY3NWI0ZWU4NjlkYTYxMzhkMDhhZjVkYzkzNmYwY2ZjZTY5MDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.giI-3JBwOqIo4TnZCsJ3rpsFd3O0PuaH6nQKknL6Xms)

Fixed

>     3. Traded instead of cancel case:
> 
> 
>     * Place an order (set 50% slippage)
> 
>     * Close the bar
> 
>     * in 15-20 seconds after an order placement cancel it --> surplus modal appears. When I close the surplus modal I can see blinking 'progress bar' modal under the Surplus one (it is auto-closed, but sometimes is noticeable).

Unsure about this one

>     4. case when a user is on Limit/TWAP/other content page:
> 
> 
>     * place a swap order. close the bar
> 
>     * go to Limit orders page
> 
>     * Open Account modal
> 
>     * Press on the 'Show progress bar' button --> looks like nothing happens, the Account modal is closed, but there is no progress bar is displayed on the Limit orders page. The bar pops-up only when go back to Swap page. Confusing.

Fixed

>     5. Also I'm noticing a slight screen blink when an order is traded with surplus less 1USD:
> 
> 
>     * place a tiny order
> 
>     * close the bar
> 
>     * wait till an order is executed and check the screen: once traded, there will be a tiny screen blink

Will check

>     6. Not sure if it is an issue, when when I reopen the bar for a ETH-flow order, I don't see the stepper above. Same when a surplus modal appears for such orders
>        ![image](https://private-user-images.githubusercontent.com/70885163/359458269-25f525b4-7030-4716-92a4-e2b9bb0c3c11.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQxNjU4ODcsIm5iZiI6MTcyNDE2NTU4NywicGF0aCI6Ii83MDg4NTE2My8zNTk0NTgyNjktMjVmNTI1YjQtNzAzMC00NzE2LTkyYTQtZTJiOWJiMGMzYzExLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIwVDE0NTMwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY4YTdmNjIyOTFkYjgyMDBiODg3MjVkNWMxOTIzMjgzZDdmYzQ0MTI5MmFlMmUyYjQ4YzM3YjA3NjhjODQyMGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.JIzHpKd2ZYDVejJMp_fM0stJjiqncpp3PE289b9PNoI)

It is an issue. Fixed

>     7. Animation stopped working (might be an intended change?):
> 
> 
>     * place 2 the same orders with a high price impact
> 
>     * wait till the 2st order is executed --> animation is stopped on the 'Price OOM' step for the 2nd order
>       ![image](https://private-user-images.githubusercontent.com/70885163/359459700-9a6f38f7-0870-4c26-8cfd-b3d61a26e328.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQxNjU4ODcsIm5iZiI6MTcyNDE2NTU4NywicGF0aCI6Ii83MDg4NTE2My8zNTk0NTk3MDAtOWE2ZjM4ZjctMDg3MC00YzI2LThjZmQtYjNkNjFhMjZlMzI4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIwVDE0NTMwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWI0YzMyY2FhMmVhNGY2MjgzYTE2MGQzNTcwNzE0N2FjMTdmNzAxMWU3MzQ0MDc5YzEzNjM5NTE2Yzg4MGEzNDYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.gK5B1Egt9xSgyYqi60iqELk16X9IqiQnMWJ-htDF1no)

That's on Michel's department. Also, we'll remove the animations so I'm not too worried

>     8. Case with 2 orders placed one after another: when the 1st order is executed with a surplus, the modal does not appear above the opened progress bar for the 2nd order
>        ![image](https://private-user-images.githubusercontent.com/70885163/359460809-3926c27e-289d-43dd-9780-eb8c053cac6c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQxNjU4ODcsIm5iZiI6MTcyNDE2NTU4NywicGF0aCI6Ii83MDg4NTE2My8zNTk0NjA4MDktMzkyNmMyN2UtMjg5ZC00M2RkLTk3ODAtZWI4YzA1M2NhYzZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIwVDE0NTMwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYwMmUxYTE1ZDRlNGFmZGVmYWFiZGZhNzg0MDJiNDgyOTEyZWQ3ZWJjOWZmNThmODI5MzliMmY3YmJkOWZhNDEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.-1ja6Upk7CvuD9mTKrDdnDKwD8hNwTizoSlFqS0smmw)

Fixed

>     9. Case with 2 orders placed one after another:
> 
> 
>     * place the 1st order, close the bar
> 
>     * place the 2nd order, leave the bar opened
> 
>     * open Account modal
> 
>     * Press on the 'Show progress' for the 1st order --> -the Account modal is closed, but I still see the progress bar for the 2nd order (not for the 1st one). WDYT about showing the 'Show progress' link to orders with closed progress bar only?

I don't see it as a problem. If progress bar open, it'll just pop out the second on top of it.

>     10. This is for @fairlighteth: when Surplus modal pops up in a mobile view, some bottom elements are hidden under the Account bar
>         ![image](https://private-user-images.githubusercontent.com/70885163/359464813-85198c17-a397-4d97-9330-3af293b5301d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQxNjU4ODcsIm5iZiI6MTcyNDE2NTU4NywicGF0aCI6Ii83MDg4NTE2My8zNTk0NjQ4MTMtODUxOThjMTctYTM5Ny00ZDk3LTkzMzAtM2FmMjkzYjUzMDFkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIwVDE0NTMwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg4YWNlZGRhZGJhMTQ2OWIzODA4YTk0YzgzNjEwMzcxMTJkZjA4N2M5YThiYWNiNjgwOWFjMDJkYWVlNWI0YjkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.oENvM6AgI0Qd2oOn-ILl3Qlo7NIgEZwVfZ4zjR8e4eM)
>         ![image](https://private-user-images.githubusercontent.com/70885163/359464887-86eed8f6-91bb-433c-9578-9ad6bd3d4bbe.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQxNjU4ODcsIm5iZiI6MTcyNDE2NTU4NywicGF0aCI6Ii83MDg4NTE2My8zNTk0NjQ4ODctODZlZWQ4ZjYtOTFiYi00MzNjLTk1NzgtOWFkNmJkM2Q0YmJlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA4MjAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwODIwVDE0NTMwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY0YWY2ZWUzNTUxYzY5MTI4ODBhMWY5OTY1YTRlZTA3ZjQ1YWU1MTUxZjYyYzAwOGI4OTA4ZjFhNWEwMzk1ZWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.8O3DJChCiojya-d50w9m2mFnq9ovN8X8ZvhKwlczZuE)
> 
> 
> Thanks

elena-zh commented:
@alfetopito , related to the case 9: 

> I don't see it as a problem. If progress bar open, it'll just pop out the second on top of it.

It does not pop out of the second on top of it.

alfetopito commented:
> @alfetopito , related to the case 9:
> 
> > I don't see it as a problem. If progress bar open, it'll just pop out the second on top of it.
> 
> It does not pop out of the second on top of it.

It does for me. Or you mean something else?

https://github.com/user-attachments/assets/95fe66e2-3f3f-4ac9-aef9-6f744a981b20

alfetopito commented:
Merging to parent to consolidate

elena-zh commented:
Changes LGTM! Thanks for addressing all these cases!
